packages/tse-backend/src/modules/strategy-execution/execution-worker.service.spec.ts

```diff
@@ -0,0 +1,136 @@
+  let arbitrageService: ArbitrageService;
+  let arbitrageStrategy: ArbitrageStrategy;
+  let marketMakingService: MarketMakingService;
+  let marketMakingStrategy: MarketMakingStrategy;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        ExecutionWorkerService,
+        {
+          provide: ArbitrageService,
+          useValue: { findRunningStrategies: jest.fn() },
+        },
+        {
+          provide: ArbitrageStrategy,
+          useValue: { start: jest.fn() },
+        },
+        {
+          provide: MarketMakingService,
+          useValue: { findRunningStrategies: jest.fn() },
+        },
+        {
+          provide: MarketMakingStrategy,
+          useValue: { start: jest.fn() },
+        },
+      ],
+    }).compile();
+
+    service = module.get<ExecutionWorkerService>(ExecutionWorkerService);
+    arbitrageService = module.get<ArbitrageService>(ArbitrageService);
+    arbitrageStrategy = module.get<ArbitrageStrategy>(ArbitrageStrategy);
+    marketMakingService = module.get<MarketMakingService>(MarketMakingService);
+    marketMakingStrategy = module.get<MarketMakingStrategy>(MarketMakingStrategy);
+  });
+
+  describe('handleCron', () => {
+    it('should execute the strategies correctly', async () => {
+      const mockArbitrageStrategies: Arbitrage[] = [
+        {
+          id: 1,
+          userId: 'user1',
+          clientId: 'client1',
+          pair: 'BTC/USD',
+          amountToTrade: 1000.00,
+          minProfitability: 0.05,
+          exchangeAName: 'ExchangeA',
+          exchangeBName: 'ExchangeB',
+          checkIntervalSeconds: 60,
+          maxOpenOrders: 5,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          status: StrategyInstanceStatus.CREATED,
+        },
+        {
+          id: 2,
+          userId: 'user2',
+          clientId: 'client2',
+          pair: 'ETH/USD',
+          amountToTrade: 500.00,
+          minProfitability: 0.03,
+          exchangeAName: 'ExchangeA',
+          exchangeBName: 'ExchangeB',
+          checkIntervalSeconds: 120,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          status: StrategyInstanceStatus.CREATED,
+        },
+      ];
+
+      const mockMarketMakingStrategies: MarketMaking[] = [
+        {
+          id: 1,
+          userId: 'user1',
+          clientId: 'client1',
+          pair: 'BTC/USD',
+          exchangeName: 'ExchangeA',
+          bidSpread: 0.01,
+          askSpread: 0.01,
+          orderAmount: 1000.00,
+          checkIntervalSeconds: 60,
+          numberOfLayers: 5,
+          priceSourceType: PriceSourceType.MID_PRICE,
+          amountChangePerLayer: 50.00,
+          amountChangeType: AmountChangeType.FIXED,
+          ceilingPrice: 50000.00,
+          floorPrice: 30000.00,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          status: StrategyInstanceStatus.CREATED,
```

Comment:
Maybe extract test data creation to a fixture of some sort

Reply:
Fixed: 4f660478